tests/jballerina-unit-test/src/test/resources/test-src/typedefs/type-definitions.bal

```diff
@@ -316,6 +316,28 @@ function testBinaryExprAssignments() {
 
 const ASSERTION_ERROR_REASON = "AssertionError";
 
+public type X1 ["x"];
+public type X2 ["x", 2, ()];
+
+function testTupleWithSingletonTypes() {
+    X1 a = returnTupleWithSingletonType();
+    X2 b = returnTupleWithSingletonType2();
+
+    X1 c = ["x"];
+    X2 d = ["x", 2, ()];
+
+    assertEquality(a, c);
+    assertEquality(b, d);
+}
+
+function returnTupleWithSingletonType() returns X1 {
+    return ["x"];
+}
+
+function returnTupleWithSingletonType2() returns X2 {
+    return ["x", 2, ()];
+}
+
```

Comment:
These are the tests for #37145

Reply:
These work for me on 2201.1.1 also. Shouldn't `returnTupleWithSingletonType` and `returnTupleWithSingletonType2` be `public` to be the same scenario as https://github.com/ballerina-platform/ballerina-lang/issues/37145?

Reply:
Yes. I will update.

Reply:
Updated. bca1c000ddcdef0dc7360bf8b70d4faf1077d9c5